packages/Utils/src/index.tsx

```diff
@@ -1,16 +1,28 @@
 import classNames from "classnames";
-import React, { FunctionComponent } from "react";
+import hoistStatics from "hoist-non-react-statics";
+import React from "react";
 import ErrorMessage from "./ErrorMessage";
+import removeExtraSpaces from "./removeExtraSpaces";
 import uniqueId from "./uniqueId";
 
-export default (ChildComponent: any) => {
-  const WithUtils: FunctionComponent = (props: object): JSX.Element => {
+const withUtils = (Component: any) => {
+  const InnerComponent = (props: object): JSX.Element => {
     const utils = {
       ErrorMessage,
       classNames,
+      removeExtraSpaces,
       uniqueId
     };
-    return <ChildComponent utils={utils} {...props} />;
+    return <Component utils={utils} {...props} />;
   };
-  return WithUtils;
+
+  InnerComponent.displayName = `withUtils(${Component.displayName ||
+    Component.name})`;
+  InnerComponent.WrappedComponent = Component;
+
+  return hoistStatics(InnerComponent, Component);
 };
+
+withUtils.displayName = "withUtils";
+
+export default withUtils;
```

Comment:
I think we should remove this HOC as it's not really doing anything but causing problems.

It's meant to save us to load some dependencies but at the end is doing the opposite, loading all the utils. Adding unnecessary logic, overcomplicating and adding modules we don't need at all and adding extra work when we do testing.

Please add a TODO and create a task to remove this and adapt all the components to get rid of this.